src/python_testing/mdns_discovery/mdns_discovery.py

```diff
@@ -0,0 +1,275 @@
+            name (str): The name of the mDNS service.
+            state_change (ServiceStateChange): The type of state change that occurred.
+
+        Returns:
+            None: This method does not return any value.
+        """
+        if state_change.value == ServiceStateChange.Added.value:
+            asyncio.ensure_future(self._query_service_info(
+                zeroconf,
+                service_type,
+                name)
+            )
+
+    async def _query_service_info(self, zeroconf: Zeroconf, service_type: str, name: str) -> None:
+        """
+        This method requests detailed information about an mDNS service, such as its address,
+        port, and other service metadata. The information is requested asynchronously.
+
+        Args:
+            zeroconf (Zeroconf): The Zeroconf instance managing the network operations.
+            service_type (str): The type of the mDNS service to query.
+            name (str): The name of the mDNS service.
+
+        Returns:
+            None: This method does not return any value. The results of the query are processed internally.
+        """
+        # Get service info
+        service_info = AsyncServiceInfo(service_type, name)
+        is_service_discovered = await service_info.async_request(zeroconf, 3000)
+
+        if is_service_discovered:
+            # Add service info to discovered services
+            for mdns_type in MdnsType:
+                if service_type == mdns_type.value:
+                    service_info_dict = self._service_info_to_dict(service_info)
+                    self._discovered_services[mdns_type.name].append(service_info_dict)
+                    break
+
+    def _service_info_to_dict(self, service_info: AsyncServiceInfo) -> Dict[str, any]:
+        """
+        Converts an AsyncServiceInfo object to a dictionary for easier access and manipulation.
+
+        This method facilitates the handling of service information by converting the AsyncServiceInfo
+        object, which contains details about an mDNS service, into a more accessible dictionary format.
+
+        Args:
+            service_info (AsyncServiceInfo): The AsyncServiceInfo object to convert.
+
+        Returns:
+            dict: A dictionary representation of the AsyncServiceInfo object.
+        """
+        service_info_dict = {
+            "service_name": service_info.name,
+            "name": service_info.get_name(),
+            "type": service_info.type,
+            "server": service_info.server,
+            "port": service_info.port,
+            "addresses": service_info.parsed_addresses(),
+            "txt_record": service_info.decoded_properties,
+            "priority": service_info.priority,
+            "interface_index": service_info.interface_index,
+            "weight": service_info.weight,
+            "host_ttl": service_info.host_ttl,
+            "other_ttl": service_info.other_ttl
```

Comment:
Please use @dataclass for this, so that it can be strongly typed, documented and referenced in types, rather than a dict which needs have string access, and thus requires constantly to look-up the keys. Data classes can be used with an `asdict` function which is very user friendly (see https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict) if you want a dict.